File: components/test/component/applications/prudence-test/mapped/data/jython/default.e.py
import sys

from java.util.concurrent.locks import ReentrantReadWriteLock
from org.restlet.ext.json import JsonRepresentation
from jython.context import *

import jython.minjson as json

# (Note that we made a small change to minjson in order to accommodate Jython.
# See line 334 there.)

# State
#
# These make sure that our state is properly stored in the context,
# so that we always use the same state, even if this script is recompiled.

def get_state_lock(conversation):
	def create_default():
		return ReentrantReadWriteLock()
	return get_context_attribute(conversation, 'jython.stateLock', create_default)

def get_state(conversation):
	def create_default():
		return {'name': 'Coraline', 'media': 'Film', 'rating': 'A+', 'characters': ['Coraline', 'Wybie', 'Mom', 'Dad']}
	return get_context_attribute(conversation, 'jython.state', create_default)

def set_state(conversation, value):
	conversation.resource.context.attributes['jython.state'] = value

def handle_init(conversation):
    conversation.addMediaTypeByName('application/json')
    conversation.addMediaTypeByName('text/plain')

def handle_get(conversation):
	r = None
	state_lock = get_state_lock(conversation)
	state = get_state(conversation)

	state_lock.readLock().lock()
	try:
		r = json.write(state)
	finally:
		state_lock.readLock().unlock()
	
	# Return a representation appropriate for the requested media type
	# of the possible options we created in handle_init()

	if conversation.mediaTypeName == 'application/json':
		r = JsonRepresentation(r)
		
	return r

def handle_post(conversation):
	# Note that we are using the minjson library to parse the entity. While
	# a simple eval() would also work, minjson.read() is safer.

	update = json.read(conversation.entity.text)
	state_lock = get_state_lock(conversation)
	state = get_state(conversation)

	# Update our state
	state_lock.writeLock().lock()
	try:
		for key in update:
			state[key] = update[key]
	finally:
		state_lock.writeLock().unlock()

	return handle_get(conversation)

def handle_put(conversation):
	# See comment in handle_post()

	update = json.read(conversation.entity.text)
	set_state(conversation, update)

	return handle_get(conversation)

def handle_delete(conversation):
	set_state(conversation, {})

	return None

File: components/test-old/component/applications/prudence-test/mapped/data/rhino/default.e.js
importClass(
	java.util.concurrent.locks.ReentrantReadWriteLock,
	org.restlet.ext.json.JsonRepresentation)

// Include the context library
document.execute('rhino/context/')

// Include the JSON library
document.execute('rhino/json2/')

// State
//
// These make sure that our state is properly stored in the context,
// so that we always use the same state, even if this script is recompiled.

function getStateLock(conversation) {
	return getContextAttribute(conversation, 'rhino.stateLock', function() {
		return new ReentrantReadWriteLock()
	})
}

function getState(conversation) {
	return getContextAttribute(conversation, 'rhino.state', function() {
		return {name: 'Coraline', media: 'Film', rating: 'A+', characters: ['Coraline', 'Wybie', 'Mom', 'Dad']}
	})
}

function setState(conversation, value) {
	conversation.resource.context.attributes.put('rhino.state', value) 
}

function handleInit(conversation) {
    conversation.addMediaTypeByName('application/json')
    conversation.addMediaTypeByName('text/plain')
}

function handleGet(conversation) {
	var r
	var stateLock = getStateLock(conversation)
	var state = getState(conversation)
	
	stateLock.readLock().lock()
	try {
		r = JSON.stringify(state)
	}
	finally {
		stateLock.readLock().unlock()
	}
	
	// Return a representation appropriate for the requested media type
	// of the possible options we created in handleInit()

	if(conversation.mediaTypeName == 'application/json') {
		r = new JsonRepresentation(r)
	}
	
	return r
}

function handlePost(conversation) {
	// Note that we are using the JSON library to parse the entity. While
	// a simple eval() would also work, JSON.parse() is safer.
	// Note, too, that we are using String() to translate the Java string
	// into a JavaScript string. In many cases this is unnecessary, but
	// in this case the JSON library specifically expects a JavaScript string
	// object.
	
	var update = JSON.parse(String(conversation.entity.text))
	var stateLock = getStateLock(conversation)
	var state = getState(conversation)
	
	stateLock.writeLock().lock()
	try {
		for(var key in update) {
			state[key] = update[key]
		}
	}
	finally {
		stateLock.writeLock().unlock()
	}
	
	return handleGet(conversation)
}

function handlePut(conversation) {
	// See comment in handlePost()

	var update = JSON.parse(String(conversation.entity.text))
	setState(conversation, update)
	
	return handleGet(conversation)
}

function handleDelete(conversation) {
	setState(conversation, {})
	
	return null
}
